drivers/flash/Kconfig.mcux

```diff
@@ -33,6 +33,40 @@ config FLASH_MCUX_FLEXSPI_NOR
 	select FLASH_HAS_DRIVER_ENABLED
 	select FLASH_MCUX_FLEXSPI
 
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	depends on FLASH_MCUX_FLEXSPI
+	depends on (CODE_FLEXSPI || CODE_FLEXSPI2)
+	select XIP
+	select CODE_DATA_RELOCATION
+	help
+	  Allows using the flash API while running in XIP.
+	  WARNING: It is possible to overwrite the running application itself.
+
+if FLASH_MCUX_FLEXSPI_XIP
+
+choice FLASH_MCUX_FLEXSPI_XIP_MEM_TARGET
+	prompt "FlexSPI drivers relocation target"
+	default FLASH_MCUX_FLEXSPI_XIP_MEM_ITCM
+	help
+	  Select the location to run the FlexSPI drivers when using
+	  the flash API.
+
+config FLASH_MCUX_FLEXSPI_XIP_MEM_ITCM
+	bool "ITCM"
+
+config FLASH_MCUX_FLEXSPI_XIP_MEM_SRAM
+	bool "SRAM"
+
+endchoice
+
+config FLASH_MCUX_FLEXSPI_XIP_MEM
+	string
+	default "ITCM" if FLASH_MCUX_FLEXSPI_XIP_MEM_ITCM
+	default "SRAM" if FLASH_MCUX_FLEXSPI_XIP_MEM_SRAM
+
```

Comment:
Would we need to adjust MPU attributes if user chooses SRAM?

Reply:
This is something I don't have experience with, please advise on how this should be done.

Do I remove `SRAM` as a target option? I added it so the choice `Kconfig` would make sense
But we can change the MPU attributes if `FLASH_MCUX_FLEXSPI_XIP_MEM_SRAM` is selected.

Reply:
I have added a PR:https://github.com/zephyrproject-rtos/zephyr/pull/34086 to fix this.